docs/spec.md

```diff
@@ -4001,49 +4001,58 @@ doesn't hold for many popular reductions. E.g. floating-point addition for
 `body` and zero for `init_values` don't actually form a monoid because
 floating-point addition is not associative.
 
-More formally, `results...[j0, ..., jR-1] = reduce(input_slices)` where:
+More formally, `results...[j0, ..., jR-1] = reduce_implicit_convert(reduce(
+input_slices_converted), type(func_outputs(body)...), type(results...)))` where:
 
 * `input_slices = inputs...[j0, ..., :, ..., jR-1]`, where `:` are inserted
   at `dimensions`.
-* `reduce(input_slices) = exec(schedule)` for some binary tree `schedule`
-  where:
+* `input_slices_converted = reduce_implicit_convert(input_slices...,
+  type(inputs...), type(func_inputs(body)...)`.
```

Comment:
Only the first half of the `func_inputs(body)`?

Reply:
agreed. fixed.